File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) 
    {
     
        int rows = matrix.size();
        int columns = matrix[0].size();
        int m = 0, n = columns - 1;
    
        while (m < rows && n >= 0)
        {
            int start = matrix[m][n];
            if(start == target) return true;
            else if(start > target) n--;
            else m++;
        }
        return false;
        
    }
};

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        m, n = len(matrix), len(matrix[0])
        left, right = 0, m * n - 1
        while left <= right:
            mid = (left + right) // 2
            pivot_element = matrix[mid // n][mid % n]
            if pivot_element == target:
                return True
            elif pivot_element < target:
                left = mid + 1
            else:
                right = mid - 1
        return False
        